testing/upload_validation.ts

```diff
@@ -594,6 +596,28 @@ function buildMetadataSchema({sdkVersion}: BuildMetadataSchemaArgs): {
       nestedResponseKey: z.string().optional(),
       credentialsLocation: z.nativeEnum(TokenExchangeCredentialsLocation).optional(),
       ...baseAuthenticationValidators,
+    }).superRefine(({requiresEndpointUrl, authorizationUrl, tokenUrl}, context) => {
+      const isRelativeUrl = (url: string) => url.startsWith('/');
+      const isAbsoluteUrl = (url: string) => url.startsWith('https://');
+      const addIssue = (property: string) => {
+        const expectedType = requiresEndpointUrl ? 'a relative' : 'an absolute';
+        context.addIssue({
+          code: z.ZodIssueCode.custom,
+          path: [property],
+          message: `${property} must be ${expectedType} URL when requiresEndpointUrl is \
+${requiresEndpointUrl ?? 'not true'}`,
+        });
+      };
+
+      if (
+        (requiresEndpointUrl && !isRelativeUrl(authorizationUrl)) ||
```

Comment:
Per the recent comment about Salesforce, if `endpointKey` is provided I think we still need to enforce an absolute url. The combination of `requiresEndpointUrl` and `endpointKey` means "use this single absolute url for the oauth flow; the token exchange response will contain the endpoint keyed by this given property".

Reply:
Updated, let me know if that looks good.